be/src/vec/functions/function_json.cpp

```diff
@@ -244,17 +244,44 @@ struct GetJsonNumberType {
         for (size_t i = 0; i < size; ++i) {
             const char* l_raw_str = reinterpret_cast<const char*>(&ldata[loffsets[i - 1]]);
             int l_str_size = loffsets[i] - loffsets[i - 1];
-
```

Comment:
better use bool do do check

Reply:
it has other common dependencies. found another way to extract common logic.